### PR TITLE
Gun Menu Update (1.5.0)

### PR DESCRIPTION
**Pull request by oqyh**
Added Custom Commands In config.json 

` "InGameGunMenuCommands": "!gunmenu,!gunsmenu,!menugun,!menuguns"`

Added Menu Customizable In Lang

### CODE REVIEW

**Comment by oqyh** [comment]
if you want to disable it make it 

"InGameGunMenuCommands": "",

i made check null first IsNullOrEmpty

File: RetakesAllocator/Menus/AdvancedGunMenu.cs [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+using static RetakesAllocatorCore.PluginInfo;
+
+namespace RetakesAllocator.AdvancedMenus;
+
+public class AdvancedGunMenu
+{
+    private RetakesAllocator retakesAllocatorInstance;
+    public AdvancedGunMenu(RetakesAllocator allocatorInstance)
+    {
+        this.retakesAllocatorInstance = allocatorInstance;
```
Comment (yonilerner) [inline RetakesAllocator/Menus/AdvancedGunMenu.cs:18]: Doesnt seem like this `allocatorInstance` is actually used, can we remove it

File: RetakesAllocator/RetakesAllocator.cs [hunk 2]
```diff
@@ -72,7 +72,13 @@ public override void Load(bool hotReload)
         {
             HandleHotReload();
         }
+        RetakesAllocator retakesAllocatorInstance = new RetakesAllocator();
```
Comment (yonilerner) [inline RetakesAllocator/RetakesAllocator.cs:75]: Dont think we should do this

File: RetakesAllocator/Menus/AdvancedGunMenu.cs [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+        var playerid = player.SteamID;
+
+        if (string.IsNullOrWhiteSpace(eventmessage)) return HookResult.Continue;
+        string trimmedMessageStart = eventmessage.TrimStart();
+        string message = trimmedMessageStart.TrimEnd();
+        string[] MenuCommands = Configs.GetConfigData().InGameGunMenuCommands.Split(',');
+
+        if (MenuCommands.Any(cmd => cmd.Equals(message, StringComparison.OrdinalIgnoreCase)))
+        {
+            var retakesAllocator = new RetakesAllocator();
```
Comment (yonilerner) [inline RetakesAllocator/Menus/AdvancedGunMenu.cs:42]: unncessary

File: RetakesAllocator/Menus/AdvancedGunMenu.cs (hunk 1 above)
Comment (oqyh) [reply RetakesAllocator/Menus/AdvancedGunMenu.cs]: blame github on that i dont it manual update to not get alot of Commits all of reds and greens x)

File: RetakesAllocator/RetakesAllocator.cs (hunk 2 above)
Comment (oqyh) [reply RetakesAllocator/RetakesAllocator.cs]: i wanna separate on tick other events from original .cs to organize

File: RetakesAllocator/Menus/AdvancedGunMenu.cs (hunk 3 above)
Comment (oqyh) [reply RetakesAllocator/Menus/AdvancedGunMenu.cs]: same thing blame github x)

**Comment by oqyh** [comment]
done :)

**Comment by oqyh** [comment]
Added

https://github.com/yonilerner/cs2-retakes-allocator/issues/45

**Comment by oqyh** [comment]
Added config "ResetStateOnGameRestart"

https://github.com/yonilerner/cs2-retakes-allocator/issues/93

**Review by yonilerner — APPROVED** [review]

**Pull request merged**